exasol_error_reporting_python/error_message_builder.py

```diff
@@ -0,0 +1,69 @@
+import re
+from typing import Any
+
+from exasol_error_reporting_python.parameters_mapper import ParametersMapper
+from exasol_error_reporting_python.placeholder_handler import PlaceholderHandler
+
+ERROR_CODE_FORMAT = "^[FEW]-[[A-Z][A-Z0-9]+(-[A-Z][A-Z0-9]+)*-[0-9]+$"
+
+
+class ErrorMessageBuilder:
+    def __init__(self, error_code: str):
+        assert re.compile(ERROR_CODE_FORMAT).match(error_code)
+
+        self._error_code = error_code
+        self._message_builder = []
+        self._mitigations = []
+        self._parameter_dict = {}
+
+    def message(self, message: str, *arguments):
+        self._message_builder.append(message)
+        self._add_parameters(message, arguments)
+        return self
+
+    def mitigation(self, mitigation: str, *arguments):
```

Comment:
you could also use key-value parameters to define the parameters

Reply:
If I understand correctly, you suggest I take the parameters as **kwargs.  if so, the way the parameters are given to the function is not  proper for this . 
For example:
https://github.com/exasol/error-reporting-java/blob/8fd5202187c45bc3c837abad8cb4bcfd5f316d27/src/test/java/com/exasol/errorreporting/ErrorMessageBuilderTest.java#L173

Reply:
that was only an idea, because **kwargs doesn't exist in java. So, we have actually more possibilities in python. And, if you look at the format function in python, that is not too uncommon to use kwargs, but we can keep it for the moment this way.